src/Microsoft.SqlTools.ServiceLayer/Rename/Requests/RenameTableInfo.cs

```diff
@@ -0,0 +1,18 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+namespace Microsoft.SqlTools.ServiceLayer.Rename.Requests
+{
+    /// <summary>
+    /// Property class for Metadata Service
+    /// </summary>
+    public class RenameTableInfo
+    {
+        public string TableName { get; set; }
+        public string OldName { get; set; }
```

Comment:
does the front end not have the Urn of the SMO object already? If not, you should consider finding out how to get it sent there when object explorer builds the tree. Otherwise this service is going to have so many special cases for all the object types it's going to get way too complicated.

Reply:
I think we can't assume that every frontend has the urn of the object already. It is true, of course, that with this approach we have many cases to support (for the alpha version only tables and columns). If we would assume that not every frontend has an object explorer, this would have to be built in the frontend. What do you think about that?

Reply:
I'm not versed enough on ADS architecture to know what the different frontends are that might use this service.

Reply:
It's true that not every provider will necessarily use URNs to identify their objects - but instead we can just assume that every provider will have a unique identifier (in our case the URN) to determine which object to update. 

So my suggestion here is to make the request this : 

```csharp
    public class RenameRequestParams
    {
        public string ObjectId { get; set; }
        public string NewName { get; set; }
    }

    public class RenameObjectRequest
    {
        public static readonly RequestType<RenameRequestParams, bool> Type = RequestType<RenameRequestParams, bool>.Create("rename/rename");
    }
```

And then for our case we'd assume (require) that ObjectId is the URN of the object, which we'd generate on the ADS side.  This will make it easily extensible for other providers since they can define what the object ID is themselves. 

Although I may also be overthinking this, if a provider wanted to implement their own rename they can just add a menu item contribution to OE for their own provider items....that would definitely make it simpler to design this since going and adding a rename feature to the core data protocol stuff would be a good chunk more work....

Let me think about it a bit and get back to you.

Reply:
I think here also of applications outside of the ADS. At the moment the method ```GetSmoObject()``` only works via the Urn. If we now change to the "ObjectId" field, we need to specify that we need the Urn to be transmitted. This means especially precise specification work, especially in case someone wants to use this project outside of ADS. So in the end we need the Urn to use the method ```GetSmoObject()```. To sum up the field ``` ObjectId``` must be the Urn in this case.
If I think from the point of view of a developer, who wants to use this library in his own project. It is not clear that the ObjectId for that service is the Urn. So for an outside developer in my opinion, it is more clear to transmit the sql object name, which I want to be renamed. Maybe in the future it makes sense to write a service method that returns the Urn for an object name. This could be used if in a second step the rename service would be replaced by an “object change service”. For me the Urn is the ObjectId, which is used in this project or the "[sqlmanagementobjects](https://github.com/microsoft/sqlmanagementobjects)" project, so it is more an identification for intern purposes.
I am not sure, if I could explain myself. If not I am sorry 😉.

Reply:
I think Charles means that any frontend has to get a reference to a SQL schema object from the tools service, whether it's from the same SQL provider that provides this rename service or some other provider that implements a rename service for its objects.
That object reference has to include an object id that can be used in these round trips and is opaque to the client. You wouldn't pass an objectid from a different provider to the rename service in this provider.

Reply:
Yeah I thought of that only to use the service "rename". It is like only calling the service without the hassle to get the object reference like a "simple method call".

Reply:
What @shueybubbles said is correct, I was just trying to think in more general terms. 

Anyways after thinking about it I think it's best just to move forward with keeping this specific to MSSQL for now. If this is a feature that we want to design in a general way later on then we'll want to be much more deliberate and plan out what that would look like - which is beyond the scope of what I think is reasonable to do at this time. We can always design that feature later and then remove this stuff at that point if needed. 

@alanrenmsft FYI - we can chat offline if you have different thoughts